worlds/cv64/__init__.py

```diff
@@ -0,0 +1,313 @@
+        if self.drac_condition == DraculasCondition.option_crystal:
+            self.total_s2s = 1
+            self.required_s2s = 1
+        elif self.drac_condition == DraculasCondition.option_specials:
+            self.total_s2s = self.options.total_special2s.value
+            self.required_s2s = int(self.options.percent_special2s_required.value / 100 * self.total_s2s)
+        else:
+            self.total_s2s = 0
+            self.required_s2s = 0
+
+        # Enable/disable character stages and branching paths accordingly
+        self.reinhardt_stages = True
+        self.carrie_stages = True
+        self.branching_stages = False
+
+        if self.options.character_stages == CharacterStages.option_reinhardt_only:
+            self.carrie_stages = False
+        elif self.options.character_stages == CharacterStages.option_carrie_only:
+            self.reinhardt_stages = False
+        elif self.options.character_stages == CharacterStages.option_both:
+            self.branching_stages = True
+
+        self.active_stage_exits = get_normal_stage_exits(self)
+
+        stage_1_blacklist = []
+
+        # Prevent Clock Tower from being Stage 1 if more than 4 S1s are needed to warp out of it.
+        if self.s1s_per_warp > 4 and not self.options.multi_hit_breakables:
+            stage_1_blacklist.append(rname.clock_tower)
+
+        # Shuffle the stages if the option is on.
+        if self.options.stage_shuffle:
+            self.active_stage_exits, self.starting_stage, self.active_stage_list = \
+                shuffle_stages(self, stage_1_blacklist)
+        else:
+            self.starting_stage = rname.forest_of_silence
+            self.active_stage_list = [stage for stage in vanilla_stage_order if stage in self.active_stage_exits]
+
+        # Create a list of warps from the active stage list. They are in a random order by default and will never
+        # include the starting stage.
+        self.active_warp_list = generate_warps(self)
+
+    def create_regions(self) -> None:
+        # Add the Menu Region.
+        reg_names = ["Menu"]
+
+        # Add every stage Region by checking to see if that stage is active.
+        reg_names.extend([name for name in get_region_names(self.active_stage_exits)])
+
+        # Add the Renon's shop Region if shopsanity is on.
+        if self.options.shopsanity:
+            reg_names.append(rname.renon)
+
+        # Add the Dracula's chamber (the end) Region.
+        reg_names.append(rname.ck_drac_chamber)
+
+        # Set up the Regions correctly
+        self.multiworld.regions.extend([Region(name, self.player, self.multiworld) for name in reg_names])
+
+        # Add the warp Entrances to the Menu Region.
+        self.get_region("Menu").add_exits(get_warp_entrances(self.active_warp_list))
+
+        for reg_name in reg_names:
+
+            # Add the Locations to all the Regions.
+            loc_names = get_region_info(reg_name, "locations")
+            if loc_names is not None:
```

Comment:
less indentation is almost always better and easier to read
```suggestion
            if loc_names is None:
                continue
```
(and then fix the indentation of the lines below it)